src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/AdsKruxObject.java

```diff
@@ -104,4 +103,27 @@ public String getKsgmntPattern(String segmentsLocalStorage) {
     return ksgmnt;
   }
 
+  /**
+   * Slice the krux segments to MAX_SEGS_NUMBER_GPT number for gpt call and set highest priority to
+   * "ph3uhzc41" segment
+   *
+   * @param segments Krux segments
+   */
+  private List<String> sliceSegsForGpt(String segments) {
+    String specialSegment = "ph3uhzc41";
+    String[] segs = segments.split(",");
+    if (segs.length > MAX_SEGS_NUMBER_GPT) {
+      if (segments.contains(specialSegment) &&
+          !Arrays.asList(segs).subList(0, MAX_SEGS_NUMBER_GPT).contains(specialSegment)) {
```

Comment:
What do you think about assigning `Arrays.asList(segs).subList(0, MAX_SEGS_NUMBER_GPT)` to a variable which you can check in this condition or return in line 123?